package tracker;

option cc_generic_services = true;

message F2TRegisterRequest
{
	required uint32 ip = 1;
	required uint32 port = 2;
	required uint32 asn = 3;
	required uint32 region = 4;
};

message F2TRegisterResponse
{
	required uint32 result = 1;
};

message F2TAddrRequest
{
	required uint32 ip = 1;
	required uint32 port = 2;
	required uint32 asn = 3;
	required uint32 region = 4;
	required uint32 level = 5;
};

message F2TAddrResponse
{
	required uint32 ip = 1;
	required uint32 port = 2;
	required uint32 result = 3;
	required uint32 level = 4;
};

message F2TUpdateStreamRequest
{
	required uint32 cmd = 1; // 0: del; 1: add 
	required uint32 level = 2;
};

message F2TUpdateStreamResponse
{
	required uint32 result = 1;
};

service EchoService
{
    rpc Register(F2TRegisterRequest) returns (F2TRegisterResponse);
	rpc GetAddr(F2TAddrRequest) returns (F2TAddrResponse);
	rpc UpdateStream(F2TUpdateStreamRequest) returns (F2TUpdateStreamResponse);
};
